src/idl_gen_json_schema.cpp

```diff
@@ -153,7 +153,12 @@ class JsonSchemaGenerator : public BaseGenerator {
         comment.append(*comment_line);
       }
       if (comment.size() > 0) {
-        code_ += "      \"description\" : \"" + comment + "\",";
+        std::string description;
+        description.reserve(comment.size());
+        if (!EscapeString(comment.data(), comment.length(), &description, true, true)) {
+          return false;
+        }
+        code_ += "      \"description\" : " + description + ",";
```

Comment:
The `EscapeString` has "append" sematic.
I think it is possible to pass the `code_` directly into this function without using the `description` string.

Reply:
Not simply: `code_` is of type `CodeWriter`, but the argument needs to be `std::string`.

Moving `EscapeString` up into the loop that constructs `comment` would not work, as `EscapeString` adds `"` prefix and suffix.

To realize your suggestion, it would require further work and a larger patch. This does not appear to be a critical code path. Would refactoring the interface to `EscapeString` serve a greater purpose in the project?

Reply:
You are right, the temporary string is needed. I forgot that `code_` isn't `std::string`.

1. It would be better change  `comment.data()` to `comment.c_str()`.
2. Call of `description.reserve(comment.size())` isn't necessary here.
We don't know how `EscapeString` works. This function would call `resize()` if it was written with the optimization of memory allocation.

Reply:
@vglavnyy I've made the changes you requested.

1. I'm' not sure why. The interface clearly takes length, so the `\0` terminator is unneeded. Both interfaces are used in this code base, that said: I just read C++11 makes `c_str` and `data` synonyms.

1. You're right, I should not try to anticipate the allocation strategy. I do know however that description will need to store a string guaranteed to be of at least the size of comment, so I don't see how this would adversely affect `resize()`. Worst case is that the reserve() call is a duplicate.

Reply:
Thank you.
1. In code of the library method `length()` usually used in pair with `c_str()` while `data()` with `size()`.

Can you apply clang-format for selected lines (156-159) in idl_gen_json_schema.cpp?
Line 157 is too long. If I right it should be like this:
```cpp
      if (comment.size() > 0) {
        std::string description;
        if (!EscapeString(comment.c_str(), comment.length(), &description, true,
                          true)) {
          return false;
        }
        code_ += "      \"description\" : " + description + ",";
```
I think this PR is ready for merge.

Reply:
I ran the tool as requested, with the assumption the style is Google. Though my line numbers did not match the ones you specified.

    clang-format --verbose -i --style=Google --lines=156:161 idl_gen_json_schema.cpp

Reply:
There is `.clang-format` file in the root of repo. It is slightly different from the --style=Google.
In any case, еру PR is ready for merge.